crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -350,10 +350,38 @@ impl<'interner> TypeChecker<'interner> {
                         );
                     }
                 }
+            // Otherwise if the object type is a mutable reference and the method is not, insert as
+            // many dereferences as needed.
+            } else if matches!(actual_type, Type::MutableReference(_)) {
+                let (object, new_type) =
+                    self.insert_auto_dereferences(method_call.object, actual_type);
+                method_call.object = object;
+                argument_types[0].0 = new_type;
             }
         }
     }
 
+    /// Insert as many dereference operations as necessary to automatically dereference a method
+    /// call object to its base value type T.
+    fn insert_auto_dereferences(&mut self, object: ExprId, typ: Type) -> (ExprId, Type) {
+        if let Type::MutableReference(element) = typ {
+            // If that didn't work, then wrap the whole expression in an `&mut`
```

Comment:
What is `that`?

Reply:
Ah, sorry. This was accidentally copied from line 337. I'll remove it.